allennlp/data/dataset_readers/dataset_utils/text2sql_utils.py

```diff
@@ -0,0 +1,137 @@
+    dataset_split: str = data['query-split']
+
+    # TODO(Mark): currently this does not filter for duplicate _sentences_
+    # which have the same sql query. Really it should, because these instances
+    # are literally identical, so just magnify errors etc. However, doing this
+    # would make it really hard to compare to previous work. Sad times.
+    for sent_info in data['sentences']:
+        # Instead, if we're using the question split,
+        # we take the split according to the individual question.
+        if use_question_split:
+            dataset_split = sent_info['question-split']
+        # We are observing the split we're meant to use for cross-validation.
+        # set the dataset split to be test. NOTE: This was _incorrect_ in the
+        # original repo, causing test leakage in datasets which used cross-validation.
+        if cross_validation_split is not None:
+            if str(cross_validation_split) == str(dataset_split):
+                dataset_bucket = "test"
+            else:
+                dataset_bucket = "train"
+        else:
+            dataset_bucket = dataset_split
+        # Loop over the different sql statements with "equivelent" semantics
+        for sql in data["sql"]:
+            sql_variables = {}
+            for variable in data['variables']:
+                sql_variables[variable['name']] = variable['example']
+
+            text_with_variables = sent_info['text'].strip().split()
+            text_vars = sent_info['variables']
+
+            query_tokens = get_tokens(text_with_variables, text_vars)
+            sql_tokens = clean_and_split_sql(sql)
+
+            sql_data = SqlData(text=query_tokens,
+                               text_with_variables=text_with_variables,
+                               sql=sql_tokens,
+                               text_variables=text_vars,
+                               sql_variables=sql_variables)
+            yield (dataset_bucket, sql_data)
+
+            # Some questions might have multiple equivelent SQL statements.
+            # By default, we just use the first one. TODO(Mark): Use the shortest?
+            if not use_all_sql:
+                break
+
+def get_split(filename: str,
+              dataset_split: str,
+              use_all_sql: bool = False,
+              use_question_split: bool = False,
+              cross_validation_split: int = None):
+    instances = []
+    with open(filename) as input_file:
+        data = json.load(input_file)
+        if not isinstance(data, list):
+            data = [data]
+        for example in data:
+            tagged_example = process_sql_data_blob(example,
+                                                   use_all_sql,
+                                                   use_question_split,
+                                                   cross_validation_split)
+            for dataset, instance in tagged_example:
+                if dataset == dataset_split:
```

Comment:
This means you're potentially reading the file multiple times to get the various splits from it?

Reply:
Yes - I haven't concretely figured out how i'm going to do this yet, because I might refactor the data to be separated  into splits, rather than have all the splits in a single file, because this would play more nicely with allennlp.